src/rtc.c

```diff
@@ -346,17 +345,38 @@ void RTC_init(hourFormat_t format, sourceClock_t source, bool reset)
   RtcHandle.Init.OutPutType = RTC_OUTPUT_TYPE_OPENDRAIN;
 #endif /* STM32F1xx */
 
+  /* Ensure backup domain is enabled before we init the RTC so we can use the backup registers for date retention on stm32f1xx baords */
```

Comment:
to be checked : does the stm32F1xx require to "Enable the power clock using `__HAL_RCC_PWR_CLK_ENABLE()`"